UnityProject/Assets/Scripts/Managers/SignalsManager.cs

```diff
@@ -0,0 +1,145 @@
+
+namespace Managers
+{
+	public class SignalsManager : NetworkBehaviour
+	{
+		private static SignalsManager signalsManager;
+		public static SignalsManager Instance
+		{
+			get
+			{
+				if (!signalsManager)
+				{
+					signalsManager = FindObjectOfType<SignalsManager>();
+				}
+
+				return signalsManager;
+			}
+		}
+
+		public List<SignalReceiver> Recivers = new List<SignalReceiver>();
+
+		/// <summary>
+		/// Called from the server as the Recivers list is only avaliable for the host and to avoid clients from cheating.
+		/// Loops through all receivers and sends the signal if they match the signal type and/or frequancy
+		/// </summary>
+		[Server]
+		public void SendSignal(SignalEmitter emitter, SignalType type, SignalDataSO signalDataSo)
+		{
+			foreach (SignalReceiver receiver in Recivers)
+			{
+				if (receiver.SignalTypeToReceive != type) return;
+
+				if (!receiver.Frequency.IsBetween(signalDataSo.MinMaxFrequancy.x, signalDataSo.MinMaxFrequancy.y)) return;
+
+				if (receiver.SignalTypeToReceive == SignalType.PING && receiver.Emitter == emitter)
+				{
+					if (signalDataSo.UsesRange) { SignalStrengthHandler(receiver, emitter, signalDataSo); break; }
+					receiver.RecieveSignal(SignalStrength.HEALTHY);
+					break;
+				}
+				//TODO (Max) : Radio signals should be sent to relays and servers.
+				if (receiver.SignalTypeToReceive == SignalType.RADIO && AreOnTheSameFrequancy(receiver, emitter))
+				{
+					if (signalDataSo.UsesRange) { SignalStrengthHandler(receiver, emitter, signalDataSo); continue; }
+					receiver.RecieveSignal(SignalStrength.HEALTHY);
+					continue;
+				}
+				//Bounced radios always have a limited range.
+				if (receiver.SignalTypeToReceive == SignalType.BOUNCED && AreOnTheSameFrequancy(receiver, emitter))
+				{
+					SignalStrengthHandler(receiver, emitter, signalDataSo);
+				}
+			}
+		}
+
+		private bool AreOnTheSameFrequancy(SignalReceiver receiver , SignalEmitter emitter)
+		{
+			return Mathf.Approximately(receiver.Frequency, emitter.Frequancy);
+		}
+
+		private void SignalStrengthHandler(SignalReceiver receiver, SignalEmitter emitter, SignalDataSO signalDataSo)
+		{
+			SignalStrength strength = GetStrength(receiver, emitter, signalDataSo.SignalRange);
+			if (strength == SignalStrength.HEALTHY) receiver.RecieveSignal(strength);
+			if (strength == SignalStrength.TOOFAR) emitter.SignalFailed();
+
+
+			if (strength == SignalStrength.DELAYED)
+			{
+				StartCoroutine(DelayedSignalRecevie(receiver.DelayTime, receiver, strength));
+				return;
+			}
+			if (strength == SignalStrength.WEAK)
+			{
+				Random chance = new Random();
+				if (DMMath.Prob(chance.Next(0, 100)))
+				{
+					StartCoroutine(DelayedSignalRecevie(receiver.DelayTime, receiver, strength));
+				}
+				emitter.SignalFailed();
+			}
+		}
+
+		private IEnumerator DelayedSignalRecevie(float waitTime, SignalReceiver receiver, SignalStrength strength)
+		{
+			yield return new WaitForSeconds(waitTime);
```

Comment:
```suggestion
			yield return new WaitForSeconds(waitTime);
```

Use WaitFor (think thats whats its called) Class instead